src/ntlm/messages/client/authenticate.rs

```diff
@@ -118,12 +118,17 @@ pub fn write_authenticate(
         ntlm_v2_hash.as_ref(),
         challenge_message.timestamp,
     )?;
-    let session_key = OsRng.gen::<[u8; SESSION_KEY_SIZE]>();
+    let mut session_key = OsRng.gen::<[u8; SESSION_KEY_SIZE]>();
     let encrypted_session_key_vec = Rc4::new(&key_exchange_key).process(session_key.as_ref());
     let mut encrypted_session_key = [0x00; ENCRYPTED_RANDOM_SESSION_KEY_SIZE];
     encrypted_session_key.clone_from_slice(encrypted_session_key_vec.as_ref());
 
-    context.flags = get_flags(context.flags, credentials);
+    context.flags = get_flags(context, credentials);
+
+    if !context.flags.contains(NegotiateFlags::NTLM_SSP_NEGOTIATE_KEY_EXCH) {
+        session_key = key_exchange_key;
+    }
```

Comment:
Nit: I understand that we re-use `key_exchange_key` unless the `NTLM_SSP_NEGOTIATE_KEY_EXCH` flag is set, in which case we generate a different session key. Instead of generating a session key that we never use and use a mutable binding, you could do that:
```rust
    let session_key = if context.flags.contains(NegotiateFlags::NTLM_SSP_NEGOTIATE_KEY_EXCH) {
        OsRng.gen::<[u8; SESSION_KEY_SIZE]>()
    } else {
        key_exchange_key
    };
```
(if blocks are expressions, and this is more idiomatic)

Reply:
Ok